addons/filesystem/functions/fnc_moduleAddFile.sqf

```diff
@@ -32,13 +33,59 @@ private _permissions = [
 	]
 ];
 
-if(_path isEqualTo "") exitWith {};
+private _isEncrypted = _module getVariable "AE3_Module_AddFile_IsEncrypted";
+private _encryptionAlgorithm = _module getVariable "AE3_Module_AddFile_EncryptionAlgorithm";
+private _encryptionKey = _module getVariable "AE3_Module_AddFile_EncryptionKey";
 
-if(_isFunction) then
+if (_path isEqualTo "") exitWith {};
+
+if (_isFunction) then
 {
 	_content = compile _content;
 };
 
+if (_isEncrypted) then 
+{
+	private _mode = "encrypt";
+
+	_content = _content splitString endl;
+
+	private _row = "";
+
+	{
```

Comment:
Why do you changed the encryption loop. I don't understand, why you now check and inititalise the encryption method for each row seperatly

Reply:
The "columnar" algorithm also encrypts the line breaks, meaning line breaks on wrong positions. That leads to disturbed text can't be decrypted because the length of the message needs to match the password and you don't know where a line would end. So partially decrypting isn't possible. That beeing said my approach was to not encrypt the complete text but line by line. Therefore I need a loop somewhere. If the loop isn't outside of the switch statement, I would need to loops, one in each case statement. As we say in germany: "Nicht schön, aber selten." Any suggestions to improve to code are welcome.

Reply:
As we say in germany: "Vorschlag zur Güte" ^^ (Not tested)
```
private _crypto_fnc;
switch (_encryptionAlgorithm) do
{
	case "caesar":
	{
		_encryptionKey = _encryptionKey call BIS_fnc_parseNumber; // needs a number
		_encryptionKey = round _encryptionKey; // needs an integer
		_crypto_fnc = AE3_armaos_fnc_encryption_caesar;
	};
	case "columnar":
	{
		_crypto_fnc = {
			params ["_encryptionKey", "_mode", "_row"];
			_row = _row regexReplace [" ", "_"];

			while {(count _encryptionKey) < 2 } do 
			{
				// min. length 2
				_encryptionKey = _encryptionKey + "_";
			};
			_crypto_fnc = AE3_armaos_fnc_encryption_columnar;
		}
		
	};
};

{
	_row = [_encryptionKey, _mode, _x] call _crypto_fnc;
	_content set [_forEachIndex, _row];
} forEach _content;
```

Reply:
Looks elegant. I've adapted your suggestion. Thank you. Please review the changes.